api/crl.go

```diff
@@ -3,30 +3,44 @@ package api
 import (
 	"encoding/pem"
 	"net/http"
+	"time"
 
 	"github.com/smallstep/certificates/api/render"
+	"github.com/smallstep/certificates/errs"
 )
 
 // CRL is an HTTP handler that returns the current CRL in DER or PEM format
 func CRL(w http.ResponseWriter, r *http.Request) {
-	crlBytes, err := mustAuthority(r.Context()).GetCertificateRevocationList()
+	crlInfo, err := mustAuthority(r.Context()).GetCertificateRevocationList()
 	if err != nil {
 		render.Error(w, err)
 		return
 	}
 
+	if crlInfo == nil {
+		render.Error(w, errs.New(http.StatusInternalServerError, "no CRL available"))
+		return
+	}
```

Comment:
Perhaps a 404 is a better response if a CA doesn't have the CRL configured.

Reply:
I've changed this to a 404 in 3dbb4aad3dc338f3d838f2c153019a84064c7eab.

The 404 case was already returned one layer down, and in general `crlInfo` won't ever be `nil` in the handler, but better safe than sorry 🙂